src/messaging/ExchangeMgr.cpp

```diff
@@ -219,15 +219,20 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
                 ec->SetMsgRcvdFromPeer(true);
             }
 
+            MessageFlags msgFlags;
+            if (isDuplicate == DuplicateMessage::kYes)
+            {
+                msgFlags.Set(MessageFlagValues::kDuplicateMessage);
+            }
             // Matched ExchangeContext; send to message handler.
-            ec->HandleMessage(packetHeader, payloadHeader, source, std::move(msgBuf));
+            ec->HandleMessage(packetHeader, payloadHeader, source, msgFlags, std::move(msgBuf));
             found = true;
             return false;
         }
         return true;
     });
 
-    if (found)
+    if (found || isDuplicate == DuplicateMessage::kYes)
```

Comment:
This does not seem right to me.

Let's say the other side sent us a message using MRP that did not expect a response.  We sent an ack and closed the exchange.

But our ack got lost, so the other side resends the message.  We land in this code, look for an existing exchange, don't find it.  What we should do is take the `sendAckAndCloseExchange` codepath below.  So it seems to me the logic should be more like:
```
if (found) 
{
  ExitNow(err = CHIP_NO_ERROR);
}
if (isDuplicate == DuplicateMessage::kNo) 
{
  // Current block that checks for initiator
}
else if (!payloadHeader.NeedsAck()) 
{
  ExitNow(err = CHIP_NO_ERROR);
}
```
or some equivalent.